### ISSUE TITLE
The implementation for MSLL metric is not correct

### ISSUE THREAD

**Issue by smkia** [ppsp-team/PyNM#22]
It seems the implementation of the MSLL metric at https://github.com/ppsp-team/PyNM/blob/joss_submission/pynm/util.py is not correct. The MSLL is computed by subtracting the log loss of the trivial model from the log loss of the trained model (see page 23 of https://gaussianprocess.org/gpml/chapters/RW2.pdf): "This loss can be standardized by subtracting the loss that would be obtained under the trivial model which predicts using a Gaussian with the mean and variance of the training data."

The current version has two problems:

1. The mean and std of the trivial model are computed on the test data (and not training data).
2. The formula for the second term (log loss of the trivial model) is not correct. Please use exactly the same form that is used for the first term `(0.5 * np.log(2 * np.pi * sigma**2) + (y_true - y_pred)**2 / (2 * sigma**2)` also for the second term.

As a hint, a possible implementation of the MSLL is available at https://github.com/amarquand/PCNtoolkit/blob/master/pcntoolkit/util/utils.py and within `compute_MSLL` function.

**Comment by deep-introspection** [comment]
@smkia This should be now adressed with this commit by @harveyaa : https://github.com/ppsp-team/PyNM/commit/60a91c1e9b89f5af9fcb99e868d9374b25882362

**Comment by deep-introspection** [comment]
Oh ok! Sorry, we created a dedicated branch for JOSS and expected that the reviewers were using this branch for the evaluation. Let us combine all our recent edits associated with the review and a push on PyPi then.

**Comment by harveyaa** [comment]
Hi @smkia, the PyPi version is updated :)

**Comment by smkia** [comment]
Hi @harveyaa, I cannot reproduce the GP results when running the tutorials in the binder. For example in tutorial 2, the MSLL values for the GP are positive. While the implementation of the MSLL function seems all right, do you have any idea why the MSLL value always remains positive (even when the fits are very well) throughout the experiments in the tutorials (try tutorial 2 on the binder for example)? The positive MSLL means the trivial model (only using the mean and variance of the training data) works better than the estimated model (then why bother to use a sophisticated method to estimate them while the mean and variance of training data do the job even better). I have tried my best to find out what went wrong but could not find the source.
